libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Generators/ActivityChecker.cs

```diff
@@ -0,0 +1,372 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using AdaptiveCards;
+using Microsoft.Bot.Builder.LanguageGeneration;
+using Microsoft.Bot.Schema;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Bot.Builder.Dialogs.Adaptive.Generators
+{
+    public static class ActivityChecker
+    {
+        /// <summary>
+        /// check the lg string result before generate an Activity.
+        /// </summary>
+        /// <param name="lgStringResult">string result from languageGenerator.</param>
+        /// <returns>Diagnostic list.</returns>
+        public static List<Diagnostic> Check(string lgStringResult)
+        {
+            JObject lgStructuredResult;
+            try
+            {
+                lgStructuredResult = JObject.Parse(lgStringResult);
+            }
+            catch
+            {
+                return new List<Diagnostic> { BuildDiagnostic("lg output is not a json object, and will fallback to string format.", false) };
+            }
+
+            return CheckStructuredResult(lgStructuredResult);
```

Comment:
This is using lgStructuredResult with is an unset value. 

IS this code passing sytlecop / fxcop?

Reply:
All the code has passed sytlecop / fxcop. So if there is something update on the rule? 
lgStructuredResult  would not be unset, because jobject parse would give the right answer or just pop up exceptions.